src/services/codefixes/helpers.ts

```diff
@@ -142,8 +134,18 @@ namespace ts.codefix {
             /*typeParameters*/ inJs ? undefined : map(typeArguments, (_, i) =>
                 createTypeParameterDeclaration(CharacterCodes.T + typeArguments!.length - 1 <= CharacterCodes.Z ? String.fromCharCode(CharacterCodes.T + i) : `T${i}`)),
             /*parameters*/ createDummyParameters(args.length, names, types, /*minArgumentCount*/ undefined, inJs),
-            /*type*/ inJs ? undefined : createKeywordTypeNode(SyntaxKind.AnyKeyword),
+            /*type*/ returnType,
             body ? createStubbedMethodBody(preferences) : undefined);
+
+        function resolveTypeFromNode(checker: TypeChecker, node: Node): TypeNode | undefined {
+            let type = checker.getTypeAtLocation(node);

```

Comment:
`getTypeAtLocation` is the right method for getting argument types, but for the contextual type there's a special method `checker.getContextualType` instead.
While we're here, `checker.getTypeAtLocation` always returns a defined result. So this can be:
```ts
const types = map(args, arg =>
    // Widen the type so we don't emit nonsense annotations like "function fn(x: 3) {"
    checker.typeToTypeNode(checker.getBaseTypeOfLiteralType(checker.getTypeAtLocation(arg))));
const names = ...same as before...;
const contextualType = checker.getContextualType(call);
const returnType = inJs ? undefined : contextualType && checker.typeToTypeNode(contextualType, call) || createKeywordTypeNode(SyntaxKind.AnyKeyword);
```

Then the following test will work:
```ts
class C {}
function f(n: number): void {}
f(new C().add(1, 2));
```